pkgs/tools/games/minecraft/optifine/update.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python3 -p python3 python3.pkgs.requests python3.pkgs.lxml
+
+from lxml import html
+import json
+import os.path
+import re
+import requests
+import subprocess
+
+def nix_prefetch_sha256(name):
+    return subprocess.run(['nix-prefetch-url', '--type', 'sha256', 'https://optifine.net/download?f=' + name], capture_output=True, text=True).stdout.strip()
+
+# fetch download page
+print('Looking up `https://optifine.net/downloads` for versions.')
+sess = requests.session()
+page = sess.get('https://optifine.net/downloads')
+tree = html.fromstring(page.content)
+
+# parse and extract main jar file names
+href = tree.xpath('//tr[@class="downloadLine downloadLineMain"]/td[@class="colMirror"]/a/@href')
+prog = re.compile('(OptiFine_)([0-9.]*)(.*)\.jar')
+result = [ prog.search(x) for x in href ]
+
+# format name, version and hash for each file
+catalogue = {}
+for i, r in enumerate(result):
+    print('[{}/{}] Cataloguing: {}'.format(i, len(result), r.group(0)) + ' ' * 8, end='\r')
+    index = r.group(1).lower() + r.group(2).replace('.', '_')
+    version = r.group(2) + r.group(3)
+    catalogue[index] = {
+        "version": version,
+        "sha256": nix_prefetch_sha256(r.group(0))
+    }
+
+# latest version should be the first entry
+if len(catalogue) > 0:
+    catalogue['optifine-latest'] = list(catalogue.values())[0]
+
+print('{} versions catalogued.'.format(len(result)) + ' ' * 32)
+
+# read previous versions
+d = os.path.dirname(os.path.abspath(__file__))
+with open(os.path.join(d, 'versions.json'), 'r') as f:
+    prev = json.load(f)
+
+# write a change summary
+print('Writing summary to versions.log... ', end='')
+with open(os.path.join(d, 'versions.log'), 'w') as f:
```

Comment:
Would be better to just print to standard output IMO, just so you dont have to remember to delete this temporary file. You will pretty much always want to commit right after running this anyway.

Reply:
Yes. I did this unfortunately since the maintainer script actually redirects stdout, so the output is "muted" and not readable. But I have found a different solution now.